crypto/provider_core.c

```diff
@@ -623,14 +637,21 @@ static int core_get_params(const OSSL_PROVIDER *prov, const OSSL_PARAM params[])
     int i;
     const OSSL_PARAM *p;
 
-    if ((p = OSSL_PARAM_locate(params, "openssl-version")) != NULL)
+    if ((p = OSSL_PARAM_locate(params, OSSL_PROV_PARAM_OPENSSL_VERSION)) != NULL)
         OSSL_PARAM_set_utf8_ptr(p, OPENSSL_VERSION_STR);
-    if ((p = OSSL_PARAM_locate(params, "provider-name")) != NULL)
+    if ((p = OSSL_PARAM_locate(params, OSSL_PROV_PARAM_PROV_NAME)) != NULL)
         OSSL_PARAM_set_utf8_ptr(p, prov->name);
 
+    if (prov->load_params != NULL) {
+        for (i = 0; prov->load_params[i].key != NULL; ++i) {
+            if ((p = OSSL_PARAM_locate(params, prov->load_params[i].key)) != NULL) {
```

Comment:
@levitte What I am doing here is not very very ugly.. Can you see a nicer way to pass it thru...
Maybe I should pass a INFOPAIR thru the load..

Reply:
It does give you an idea of the sort of flow that needs to happen for the self tests, even if its not quite right yet.

Reply:
Yeah, I see your point...  But hang on, I think this can be done in a different manner, which would also mimic the way application will have this loaded.  How about, instead of using all the `OSSL_PROVIDER_` functions, you constructed a complete configuration file in a buffer, and then loaded it via a mem_buf BIO with `CONF_modules_load`?

Reply:
The coolest would be to do that with a separate library context...  but then we need to have a new variant of `CONF_modules_load` that takes a `OPENSSL_CTX *` parameter.  We haven't set a goal to have the library context used everywhere for 3.0.0, so it doesn't have to be the highest priority, but sure would be nice...  enough to make a mental note.